### PR TITLE
[DNM] jewel: rgw: anonymous user error code of getting object is not consistent with SWIFT 

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/19177

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@mdw-at-linuxbox could you take a look?

**Comment by smithfarm** [comment]
Jenkins retest this please (logs gone)

**Comment by smithfarm** [comment]
Jenkins re-test this please (still no logs)

**Comment by mdw-at-linuxbox** [comment]
Your change looks harmless.  Does this enable something to pass a new test?

I believe you'll need to rebase this to bring it to a testable state.

**Comment by smithfarm** [comment]
@mdw-at-linuxbox Thanks - giving it a try.

**Comment by smithfarm** [comment]
```
rgw/rgw_rest_swift.cc: In member function ‘virtual int RGWGetObj_ObjStore_SWIFT::verify_permission()’:
rgw/rgw_rest_swift.cc:1097:10: error: ‘struct req_state’ has no member named ‘auth_identity’
   if (s->auth_identity->is_anonymous() && op_ret == -EACCES) {
          ^
rgw/rgw_rest_swift.cc:1102:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
Makefile:24861: recipe for target 'rgw/librgw_la-rgw_rest_swift.lo' failed
```

**Comment by smithfarm** [comment]
@mdw-at-linuxbox I recently found out that Jenkins merges the PR on top of the latest target branch (jewel in this case) for each test, so rebasing is not necessary.

Jenkins re-test this please

File: src/rgw/rgw_rest_swift.cc [hunk 1]
```diff
@@ -1087,6 +1087,20 @@ void RGWCopyObj_ObjStore_SWIFT::send_response()
   }
 }
 
+int RGWGetObj_ObjStore_SWIFT::verify_permission()
+{
+  op_ret = RGWGetObj_ObjStore::verify_permission();
+
+  /* We have to differentiate error codes depending on whether user is
+   * anonymous (401 Unauthorized) or he doesn't have necessary permissions
```
Comment (amitkumar50) [inline src/rgw/rgw_rest_swift.cc]: @smithfarm (NIT) I believe to replace doesn't with does.
Your thoughts?

Comment (smithfarm) [inline src/rgw/rgw_rest_swift.cc]: @amitkumar50 Yes, please do not review backport PRs. Only review if the PR is targeting master branch. Thanks!

**Comment by smithfarm** [comment]
Jenkins re-test this please (logs are gone)

**Comment by smithfarm** [comment]
Jewel is EOL

**Pull request closed without merging**